### ISSUE TITLE
GAP crashes in --cover mode when reading certain larger data files

### ISSUE THREAD

**Issue by Stefan-Kohl** [gap-system/gap#685]
The problem with crashes in --cover mode when reading certain larger data files already reported
for GAP 4.8.2 persists in GAP 4.8.3. A way to trigger this problem which avoids having to attach a
suitable data file to this issue is as follows:
```

  bin/gap.sh --cover /user/GAP/log/Profile-Test.gz
  gap> LoadPackage("rcwa");
  gap> data := LoadDatabaseOfGroupsGeneratedBy3ClassTranspositions(9);; 
  Speicherzugriffsfehler (Speicherabzug geschrieben)


```

**Comment by ChrisJefferson** [comment]
Confirmed. The problem is files with > 65,536 lines, but at the least GAP should still work, even if profiling does not.

**Comment by ChrisJefferson** [comment]
@Stefan-Kohl : Out of interest, do you know how long the longest file in the rcwa package is?

**Comment by Stefan-Kohl** [comment]
The longest data files in RCWA are rcwa/data/3ctsgroups9/mods.g (8460222 bytes / 117600 lines)
and  rcwa/data/3ctsgroups9/sizespos.g (8350893 bytes / 116324 lines). The longest source file
rcwa/lib/rcwamap.gi is much shorter (still somewhat below 10000 lines).

**Comment by ChrisJefferson** [comment]
Have found the issue, will make a fix soon (although it won't get into 4.8.3).

With this change GAP will ignore any lines after the 65,535th line of a file (for profiling, obviously the file will still load correctly!), but at least it won't crash, and other files will be handled correctly.

**Comment by Stefan-Kohl** [comment]
O.k., fine. -- Profiling data files is not necessary anyway, I think.

**Comment by ChrisJefferson** [comment]
This is now fixed in #659. That fixed @Stefan-Kohl 's previous issue, this new issue I added on top.

(It was good to catch this, I thought I'd fixed this before, but it turns out GAP gets upset if you try to enter the break loop while it is half-way through parsing a function, as this should never happen. Now I skip entering the break loop).

Unfortunately I think this patch has missed 4.8.3, but hopefully will be merged shortly.
